app/queries/get_duplicate_matches.rb

```diff
@@ -10,21 +10,21 @@ def self.call
           submitted_at
         FROM application_forms application_details
         JOIN(
-          SELECT TRIM(UPPER(application_forms.last_name)) last_name, application_forms.date_of_birth, REPLACE(UPPER(application_forms.postcode), ' ', '') postcode
+          SELECT TRIM(UPPER(application_forms.last_name)) last_name, application_forms.date_of_birth, COALESCE(REPLACE(UPPER(application_forms.postcode), ' ', ''), '') postcode
           FROM application_forms
           WHERE application_forms.previous_application_form_id IS NULL
           GROUP BY TRIM(UPPER(application_forms.last_name)), application_forms.date_of_birth, REPLACE(UPPER(application_forms.postcode), ' ', '')
```

Comment:
When I tested it with nulls it worked perfectly but didn't find matches that differed only by whitespace, e.g. (' ' and NULL). I think you need the `COALESCE` in the `GROUP BY` here and in the main `SELECT DISTINCT` clause above too.

Reply:
good spot. We've fixed and added a test for it.